src/microbe_stage/MembraneWaterRipple.cs

```diff
@@ -0,0 +1,666 @@
+            waterMaterial.SetShaderParameter(rippleStrengthParam, RippleStrength);
+            waterMaterial.SetShaderParameter(timeOffsetParam, 0.0f);
+            waterMaterial.SetShaderParameter(movementSpeedParam, 0.0f);
+            waterMaterial.SetShaderParameter(movementDirectionParam, Vector2.Zero);
+            waterMaterial.SetShaderParameter(phaseParam, 0.2f);
+            waterMaterial.SetShaderParameter(attenuationParam, 0.9998f);
+            waterMaterial.SetShaderParameter(stillnessFactorParam, 0.0f);
+            waterMaterial.SetShaderParameter(pastPositionsParam, godotPastPositions);
+            waterMaterial.SetShaderParameter(pastPositionsCountParam, 0);
+            waterMaterial.SetShaderParameter(membraneRadiusParam, 5.0f);
+            currentCamera = GetViewport().GetCamera3D();
+            waterPlane.Visible = false;
+
+            // If we got here everything is properly initialized :)
+            isInitialized = true;
+        }
+    }
+
+    public override void _Process(double delta)
+    {
+        // Skip if not initialized or missing required components
+        if (!isInitialized || !EnableEffect || positionTarget == null)
+            return;
+
+        // Clamp delta to further prevent jittering
+        float clampedDelta = MathF.Min((float)delta, MAX_DELTA_TIME);
+        currentCamera = GetViewport().GetCamera3D();
+
+        // Updates camera and visibility
+        visibilityCheckTimer += clampedDelta;
+        if (visibilityCheckTimer >= VisibilityCheckInterval)
+        {
+            visibilityCheckTimer = 0.0f;
+            UpdateCameraCache();
+            isCurrentlyVisible = IsVisible();
+
+            // Toggle water plane visibility
+            if (waterPlane.Visible != (isCurrentlyVisible && isEffectEnabled))
+            {
+                waterPlane.Visible = isCurrentlyVisible && isEffectEnabled;
+            }
+        }
+
+        if (!isCurrentlyVisible || !isEffectEnabled)
+            return;
+
+        if (positionTarget.IsInsideTree())
+        {
+            Transform3D transform = new Transform3D(Basis.Identity,
+                new Vector3(positionTarget.GlobalPosition.X,
+                    positionTarget.GlobalPosition.Y + VerticalOffset,
+                    positionTarget.GlobalPosition.Z));
+            waterPlane.GlobalTransform = transform;
+        }
+
+        UpdateRippleEffect(clampedDelta);
+        float timeScale = CalculateTimeScale();
+        timeAccumulator += clampedDelta * timeScale;
+        waterMaterial.SetShaderParameter(timeOffsetParam, timeAccumulator);
+        UpdateMovementParameters(clampedDelta);
+    }
+
+    /// <summary>
+    ///   Updates the target and size from an external source or initializes the effect.
+    /// </summary>
+    public void UpdateTarget(Node3D? target, float radius)
+    {
+        // Early return if not properly initialized
+        if (!isInitialized)
+            return;
```

Comment:
Seems like the Membrane will not retry this call. So again, it is incorrect to silently ignore the error as then this is left with inconsistent state in regards to the actual data source.